### DISCUSSION TITLE
SBT fails to properly resolve Maven timestamp based snapshots

### DISCUSSION THREAD

**Discussion by leszekgruchala** [sbt/sbt#6892] [Q&A] [answered]
Hey, I am using SBT 1.6.2.
I publish `my-internal-library` to Nexus (Sonatype) Maven style internal, remote repo using `sbt-aether-deploy` SBT plugin, so each snapshot jar is published with a unique timestamp.
I see in Nexus they are correctly published.

I have 2 problems when I `use` this library in my project:

1. Till now I depended on `7.1.0-SNAPSHOT` version. I want to depend now on `7.1.0-20220505.081738-4`. When I do `sbt update` the dependency is found and downloaded, but 
`dependencyTree` shows me that my unique version is evicted and SBT still uses `7.1.0-SNAPSHOT`. Part of the output:

```
[info]   |+-com.company:my-internal-library_2.13:7.1.0-SNAPSHOT [S]
...
[info]   | +-com.company:my-internal-library_2.13:7.1.0-20220505.081738-4 (evicte..
```

2. `my-internal-library` is configured with `ThisBuild / versionScheme := Some("early-semver")`. When I publish a new major version like `8.0.0-20220505.142831-2`, I see in Nexus both `8.0.0-SNAPSHOT` and `8.0.0-20220505.142831-2` versions. Which is correct. But when I try to depend on `8.0.0-20220505.142831-2` on the project, I get this error:

```
[error]         * com.company:my-internal-library_2.13:8.0.0-SNAPSHOT (early-semver) is selected over 8.0.0-20220505.142831-2
```

After searching half the internet, I've tried to add `addMavenResolverPlugin` in `plugins.sbt` but sbt fails to start now:
```
[info] Resolved  dependencies
[warn] 
[warn]  Note: Some unresolved dependencies have extra attributes.  Check that these dependencies exist with the requested attributes.
[warn]          org.scala-sbt:sbt-maven-resolver:1.6.2 (sbtVersion=1.0, scalaVersion=2.12)
[warn] 
[warn]  Note: Unresolved dependencies path:
[error] sbt.librarymanagement.ResolveException: Error downloading org.scala-sbt:sbt-maven-resolver;sbtVersion=1.0;scalaVersion=2.12:1.6.2
[error]   Not found
[error]   Not found
[error]   not found: https://repo1.maven.org/maven2/org/scala-sbt/sbt-maven-resolver_2.12_1.0/1.6.2/sbt-maven-resolver-1.6.2.pom
[error]   not found: /Users/lgr/.ivy2/local/org.scala-sbt/sbt-maven-resolver/scala_2.12/sbt_1.0/1.6.2/ivys/ivy.xml
[error]   not found: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.scala-sbt/sbt-maven-resolver/scala_2.12/sbt_1.0/1.6.2/ivys/ivy.xml
[error]   not found: https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt-maven-resolver/scala_2.12/sbt_1.0/1.6.2/ivys/ivy.xml
```

What can I do, to have SBT use a unique snapshot version when it is defined?

Thank you for your help

**Comment by eed3si9n** [comment, marked as answer]
I suggest opening a GitHub issue on coursier/coursier to build a consensus around how this should be handled, and if you need this feature for your work, contribute the fix to Coursier (or contract someone who would).

**Reply by leszekgruchala** [reply]
Ok. Thanks for pointing out the right direction.

**Reply by leszekgruchala** [reply]
And what about addMavenResolverPlugin? Is this still supported?
